### PR TITLE
Lovelace markdown card - fix race condition, add variables to template

### PR DESCRIPTION
**Pull request by thomasloven**
There was a race condition for the connection/disconnection to the websocket for the templates in markdown card. This should make sure things happen in the right order when redrawing the card.

I also added the card config as variables to the template. This allows for:

```yaml
type: entity-filter
entities:
  - light.bed_light
  - light.ceiling_lights
  - light.kitchen_lights
card:
  type: markdown
  content: |
    The lights that are on are:
    {% for l in entities %}
      - {{ l.entity }}
    {%- endfor %}
```

### CODE REVIEW

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 1]
```diff
@@ -91,6 +91,7 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
         {
           template: this._config.content,
           entity_ids: this._config.entity_id,
+          variables: this._config,
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-markdown-card.ts]: Can we make this 
```suggestion
          variables: {config: this._config},
```

**Comment by balloob** [comment]
Please do not forget to open pull request for the documentation

**Comment by balloob** [comment]
Note that because of the way the filter card currently works, it will set a new config on each `hass` change, even if the configuration is the same. Because of the way the markdown card works, it means that you will end up unsubscribing and resubscribing **on every state change.**

**Comment by balloob** [comment]
This can be merged when a docs PR has been linked.

**Comment by thomasloven** [comment]
Done.

I'll see what I can do about the entity-filter card...

**Pull request merged**